Libplanet/Store/FileStore.cs

```diff
@@ -134,6 +134,30 @@ public override long AppendIndex(HashDigest<SHA256> hash)
             }
         }
 
+        public override bool DeleteIndex(HashDigest<SHA256> hash)
+        {
+            bool found = false;
+            using (var writer = new MemoryStream())
+            {
+                foreach (var idx in IterateIndex())
+                {
+                    if (!found && idx == hash)
+                    {
+                        found = true;
+                    }
+                    else
+                    {
+                        writer.Write(
+                            idx.ToByteArray(), 0, HashDigest<SHA256>.Size);
+                    }
+                }
+
+                File.WriteAllBytes(_indexPath, writer.ToArray());
```

Comment:
It doesn't seem very scalable to buffer the whole index on the memory.  It should be fixed in the future.  It also would be good if we write a `FIXME` comment here.